### DISCUSSION TITLE
Stopping for robot.drive

### DISCUSSION THREAD

**Discussion by anya-codes** [pybricks/support#1238] [Q&A] [answered]
Hi,
My team and I are trying to use robot.drive() and robot.stop() in our code but when we use robot.stop() it only seems to have the stop type of coast, not hold. This causes our robot to be really inaccurate.

For example, we are trying to do line detection (with a SPIKE Prime Robot) and we are starting the motors using drive, then looking for the line, and then stopping the motors. But when we tried this, the robot stops 2-3 inches past the line.

Is there some way for the to hold the motors using robot.stop()?

**Comment by laurensvalk** [comment]
If you want an instant stop you can drive 0 mm, and then coast/brake/hold as you like, e.g.

```python
# Use drive() to get to the line as you already do and then
robot.straight(0)
```
This will hold by default, but you can specify brake if you prefer.

**Comment by anya-codes** [comment]
Thanks.
That works better than robot.stop() but still not very well.

Just wondering, is there a specific reason why robot.stop() doesn't have hold or you just haven't implemented it yet?

**Reply by laurensvalk** [reply, marked as answer]
Can you describe what isn’t working well and what you’d ideally like to do?

You can also use drive(0, 0) with the code you already have, and add a short loop that waits for the speed to truly hit zero.

This gives you control over acceleration and deceleration as usual, instead of forcing a stop.
